spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md

```diff
@@ -95,42 +138,42 @@ interface ErrorReceipt {
 
 ### Store Paths
 
-#### Upgrade Sequence Path
+#### Upgrade Channel Path
 
-The upgrade sequence path is a public path that stores the current sequence of the upgrade attempt. The sequence will increment with each attempted upgrade on the given channel. The sequence will be used to ensure that different error receipts referring to different upgrade attempts do not interfere with each other.
+The chain must store the proposed upgrade upon initiating an upgrade. The proposed upgrade must be stored in the public store. It may be deleted once the upgrade is successful or has been aborted.
 
 ```typescript
-function channelUpgradeSequencePath(portIdentifier: Identifier, channelIdentifier: Identifier) Path {
-    return "channelUpgrades/upgradeSequence/ports/{portIdentifier}/channels/{channelIdentifier}"
-}
+function channelUpgradePath(portIdentifier: Identifier, channelIdentifier: Identifier): Path {
+    return "channelUpgrades/upgrades/ports/{portIdentifier}/channels/{channelIdentifier}"
+ }
 ```
 
-The upgrade sequence MUST also have a verification method so that chains can prove the upgrade sequence on the counterparty for the given channel upgrade.
+The upgrade path has an associated verification of membership method added to the connection interface so that a counterparty may verify that chain has stored the provided counterparty upgrade.
 
 ```typescript
-// Connection VerifyChannelUpgradeSequence method
-function verifyChannelUpgradeSequence(
-  connection: ConnectionEnd,
-  height: Height,
-  proof: CommitmentProof,
-  counterpartyPortIdentifier: Identifier,
-  counterpartyChannelIdentifier: Identifier,
-  sequence: uint64
+// Connection VerifyChannelUpgrade method
+function verifyChannelUpgrade(
+    connection: ConnectionEnd,
+    height: Height,
+    proof: CommitmentProof,
+    counterpartyPortIdentifier: Identifier,
+    counterpartyChannelIdentifier: Identifier,
+    upgrade: Upgrade
 ) {
     clientState = queryClientState(connection.clientIdentifier)
-    path = applyPrefix(connection.counterpartyPrefix, channelUpgradeSequencePath(counterpartyPortIdentifier, counterpartyChannelIdentifier))
-    return verifyMembership(clientState, height, 0, 0, proof, path, sequence)
+    path = applyPrefix(connection.counterpartyPrefix, channelUpgradePath(counterpartyPortIdentifier, counterpartyChannelIdentifier))
+    return verifyMembership(clientState, height, 0, 0, proof, path, upgrade)
 }
 ```
 
-#### Restore Channel Path
+#### CounterpartyLastPacketSequence Path
 
-The chain must store the previous channel end so that it may restore it if the upgrade handshake fails. This may be stored in the private store.
+The chain must store the counterparty's last packet sequence on `startFlushUpgradeHandshake`. This will be stored in the `counterpartyLastPacketSequence` path on the private store/
```

Comment:
Does not need to be provable to counterparty. This is a moot point for current SDK since all storage is provable, but they are apparently changing this, in which case we can move any storage set in private store in the specs to this non-provable store in the sdk